l1-verifier/src/Trollup.sol

```diff
@@ -26,12 +71,17 @@ contract Trollup is Verifier {
             verifyTx(l2Block[i]);
         }
 
+        // If we have a valid block, process all transactions
+        for (uint256 i = 0; i < BLOCK_SIZE; ++i) {
```

Comment:
This loop is duplicated from line 70 above. Maybe put the call to `processTx` inside `verifyTx`?